hexer/H3Path.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <vector>
+
+#include "export.hpp"
+#include "Mathpair.hpp"
+#include "exception.hpp"
+
+#include <h3/include/h3api.h>
+
+namespace hexer
+{
+
+class H3Grid;
+
+using DirEdge = H3Index;
+
+enum H3Orientation
+{
+    H3CLOCKWISE,     // Hole
+    H3ANTICLOCKWISE  // Outer
+};
+
+class HEXER_DLL H3Path
+{
+public:
+    H3Path(H3Grid *m_grid, H3Orientation dir, const CoordIJ hex) :
+        m_grid(m_grid), m_parent(NULL), m_direction(dir),
+        m_root_hex(hex)
+    {}
+
+    ~H3Path()
+    {
+        for (std::vector<H3Path*>::size_type i = 0; i < m_children.size(); ++i)
```

Comment:
If you have to do `delete` like this, then you could do something else, like maybe this should just be a `vector<H3Path>`. But it's also strange because there is a list of paths in H3Grid. We should look at this carefully.